src/pages/settings/Security/AddDelegate/ConfirmDelegatePage.tsx

```diff
@@ -74,6 +72,13 @@ function ConfirmDelegatePage({route}: ConfirmDelegatePageProps) {
                 onPress={() => Navigation.navigate(ROUTES.SETTINGS_DELEGATE_ROLE.getRoute(login, role))}
                 shouldShowRightIcon
             />
+
+            {isValidateCodeActionModalVisible && (
```

Comment:
Coming from https://github.com/Expensify/App/issues/51273, Here, when we close the modal from the child component, the `isValidateCodeActionModalVisible` value on the parent component doesn’t get updated. As a result, when we navigate back and try to open the modal again, the button won’t trigger it because the value remains true on the parent.

More context on this in the proposal: https://github.com/Expensify/App/issues/51273#issuecomment-2429827039